src/babylon.scene.ts

```diff
@@ -1788,24 +1788,16 @@ module BABYLON {
                 if (!clickInfo.ignore) {
                     if (!clickInfo.hasSwiped) {
                         if (clickInfo.singleClick && this.onPointerObservable.hasSpecificMask(PointerEventTypes.POINTERTAP)) {
-                            let type = PointerEventTypes.POINTERTAP;
-                            let pi = new PointerInfo(type, evt, pickResult);
-                            this._setRayOnPointerInfo(pi);
-                            this.onPointerObservable.notifyObservers(pi, type);
+                            type = PointerEventTypes.POINTERTAP;
                         }
                         if (clickInfo.doubleClick && this.onPointerObservable.hasSpecificMask(PointerEventTypes.POINTERDOUBLETAP)) {
-                            let type = PointerEventTypes.POINTERDOUBLETAP;
-                            let pi = new PointerInfo(type, evt, pickResult);
-                            this._setRayOnPointerInfo(pi);
-                            this.onPointerObservable.notifyObservers(pi, type);
+                            type = PointerEventTypes.POINTERDOUBLETAP;

```

Comment:
that does not work :)
There is no "else" here between the two blocks

Reply:
That could be a function though to help reduce code complexity

Reply:
Hmm, Should it be able to fire multiple observers per run? 
I assumed it would either run a singleClick, doubleClick or standard pointerUp event type
http://playground.babylonjs.com/#UM43F8

Reply:
That's a good point. Then we should add a else to make sure and do heavy testing